packages/dds/merge-tree/src/client.ts

```diff
@@ -755,6 +755,12 @@ export class Client {
             return true;
         });
 
+        const { currentSeq, clientId } = this.getCollabWindow();
+        const fasterComputedPosition = this.mergeTree.getPosition(segment, currentSeq, clientId, localSeq);
+        if (fasterComputedPosition !== segmentPosition) {
+            this.mergeTree.getPosition(segment, currentSeq, clientId, localSeq);
+        }
+        assert(fasterComputedPosition === segmentPosition, "Expected fast-path computation to match result from walk all segments");
```

Comment:
before merging, I intend to refactor this a bit into a helper client class that the resetPendingOpToSpec as well as the fuzz tests run (having a "known correct" fallback implementation to compare against the fast-path is generally quite useful and allows earlier/more diagnosable failures)

Reply:
done